src/regex.c

```diff
@@ -5,150 +5,155 @@
 
 #include "internal.h"
 
-V7_PRIVATE enum v7_err regex_xctor(struct v7 *v7, struct v7_val *obj,
-                                   const char *re, size_t re_len,
-                                   const char *fl, size_t fl_len) {
-  if (NULL == obj) obj = v7_push_new_object(v7);
-  v7_init_str(obj, re, re_len, 1);
-  v7_set_class(obj, V7_CLASS_REGEXP);
-  obj->v.str.prog = NULL;
-  obj->fl.fl.re = 1;
-  while (fl_len) {
-    switch (fl[--fl_len]) {
-      case 'g':
-        obj->fl.fl.re_flags |= SLRE_FLAG_G;
-        break;
-      case 'i':
-      obj->fl.fl.re_flags |= SLRE_FLAG_I;
-        break;
-      case 'm':
-        obj->fl.fl.re_flags |= SLRE_FLAG_M;
-        break;
+V7_PRIVATE val_t to_string(struct v7 *, val_t);
+
+static val_t Regex_ctor(struct v7 *v7, val_t this_obj, val_t args) {
+  long argnum = v7_array_length(v7, args);
+  if (argnum > 0) {
+    val_t ro = to_string(v7, v7_array_at(v7, args, 0));
+    size_t re_len, flags_len = 0;
+    const char *re = v7_to_string(v7, &ro, &re_len), *flags = NULL;
+    struct slre_prog *p = NULL;
+    struct v7_regexp *rp;
+
+    (void)this_obj;
+    if (argnum > 1) {
+      val_t fl = to_string(v7, v7_array_at(v7, args, 1));
+      flags = v7_to_string(v7, &fl, &flags_len);
     }
-  }
-  obj->v.str.lastIndex = 0;
-  return V7_OK;
-}
+    if (slre_compile(re, re_len, flags, flags_len, &p, 1) != SLRE_OK ||
+        p == NULL) {
+      throw_exception(v7, "Error", "Invalid regex");
```

Comment:
I noticed that V8 throws a `SyntaxError` exception in this case. Should we do the same?